### DISCUSSION TITLE
Deploy preferred config to multiple machines automatically

### DISCUSSION THREAD

**Discussion by R0NAM1** [valinet/ExplorerPatcher#3379] [Q&A]
I came across this software trying to force the Windows 10 Start menu but keep the Windows 11 Taskbar and it works well, but the issue is I plan on deploying this to multiple machines from a SysPrep image and it seems the properties are reset during that.

How can we save the configuration to a file, or back it up in a file and then restore it automatically, preferably with Powershell.

This would also be good if you wanted historical data or multiple configs you can swap out at any time.

**Comment by R0NAM1** [comment]
I've discovered settings.reg, so I can just force the registry keys on machine setup or user login.

**Comment by pyrates999** [comment]
If you open up the EP properties application, and go to about.  You have the export settings feature and import settings feature.

**Comment by R0NAM1** [comment]
I was aware of that, I just needed a way to apply them with Powershell on user login. 

I assume after running
`taskkill /f /im explorer.exe`
`start explorer.exe`

Will start explorer with the correct settings? (I haven't tried this yet, to lazy to boot a VM and no work for a while.)

**Comment by pyrates999** [comment]
You could try exporting the settings that are in the registry:

* "Computer\HKEY_CURRENT_USER\Software\ExplorerPatcher"
* "HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\ExplorerPatcher"

**Comment by R0NAM1** [comment]
That's what I'll do, and I'll just have them apply on login through GPO. This combined with https://schneegans.de/windows/unattend-generator/ allows a Win 11 Taskbar with a Win10 start menu, and it's just an app list! Which is nice.

**Comment by pyrates999** [comment]
awesome :)

**Comment by pyrates999** [comment]
Just remember the settings used can change between releases.  So update the settings in the registry for your install script when new releases come out.  And if this is work computers, be careful because EP requires administrative privileges to install and update.
